pkg/engine/internal/scheduler/wire/frame_protocol.go

```diff
@@ -0,0 +1,14 @@
+package wire
+
+import (
+	"io"
+)
+
+// FrameProtocol handles reading and writing frames over a connection.
+type FrameProtocol interface {
```

Comment:
nit: do we need the interface here? 

It looks like we only ever use ProtobufProtocol, which would mean that:

* We could remove the FrameProtocol interface 
* `NewHTTP2Listener` doesn't need to accept an implementation of the interface, and can instead create its own `ProtobufProtocol` 

This would help keep the package API a little more slimmed down.